File: Vitality.Website/Areas/Global/Models/Question.cs
using System.Collections.Generic;
using Glass.Mapper.Sc.Configuration;
using Glass.Mapper.Sc.Configuration.Attributes;

namespace Vitality.Website.Areas.Global.Models
{
    public class Question : SitecoreItem
    {
        [SitecoreField(Setting = SitecoreFieldSettings.DontLoadLazily)]
        public KeyValuePair ControlType { get; set; }
        public string DefaultValue { get; set; }
        [SitecoreField(Setting = SitecoreFieldSettings.DontLoadLazily)]
        public KeyValuePair Key { get; set; }
        public string Label { get; set; }
        public string Placeholder { get; set; }
        [SitecoreField(Setting = SitecoreFieldSettings.DontLoadLazily)]
        public IEnumerable<KeyValuePair>  RelatedData { get; set; }
        [SitecoreField(Setting = SitecoreFieldSettings.DontLoadLazily)]
        public IEnumerable<FieldValidator> Validators { get; set; }
        [SitecoreField(Setting = SitecoreFieldSettings.DontLoadLazily)]
        public string Mask { get; set; }
    }
}

File: Vitality.Website/src/ts/app/Models/question.ts
import { FieldValidator } from "./field-validator";
export class Question<T>{
    value: T;
    basedOnKey: string;
    basedOnValue: number;
    key: string;
    label: string;
    placeholder: string;
    validators: FieldValidator[];
    onValueChange : any;
    controlType: string;
    isHidden: boolean;
    relatedData: { key: string, value: string }[] = [];
    scrollTo: string;
    selectedValues: Array<string> = [];
}
